src/lsdb/loaders/dataframe/from_dataframe_utils.py

```diff
@@ -0,0 +1,101 @@
+from typing import List, Tuple
+
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+from dask import delayed
+from hipscat.catalog import PartitionInfo
+from hipscat.pixel_math import HealpixPixel
+from hipscat.pixel_math.hipscat_id import HIPSCAT_ID_COLUMN
+
+from lsdb.dask.divisions import get_pixels_divisions
+
+
+def _generate_dask_dataframe(
+    pixel_dfs: List[pd.DataFrame], pixels: List[HealpixPixel]
+) -> Tuple[dd.DataFrame, int]:
+    """Create the Dask Dataframe from the list of HEALPix pixel Dataframes
+
+    Args:
+        pixel_dfs (List[pd.DataFrame]): The list of HEALPix pixel Dataframes
+        pixels (List[HealpixPixel]): The list of HEALPix pixels in the catalog
+
+    Returns:
+        The catalog's Dask Dataframe and its total number of rows.
+    """
+    schema = pixel_dfs[0].iloc[:0, :].copy()
+    divisions = get_pixels_divisions(pixels)
+    delayed_dfs = [delayed(df) for df in pixel_dfs]
+    ddf = dd.from_delayed(delayed_dfs, meta=schema, divisions=divisions)
+    return ddf if isinstance(ddf, dd.DataFrame) else ddf.to_frame(), len(ddf)
+
+
+def _append_partition_information_to_dataframe(dataframe: pd.DataFrame, pixel: HealpixPixel) -> pd.DataFrame:
+    """Append partitioning information to a HEALPix dataframe
+
+    Args:
+        dataframe (pd.Dataframe): A HEALPix's pandas dataframe
+        pixel (HealpixPixel): The HEALPix pixel for the current partition
+
+    Returns:
+        The dataframe for a HEALPix, with data points and respective partition information.
+    """
+    dataframe["Norder"] = pixel.order
```

Comment:
You're mixing use of `"Norder"` and `PartitionInfo.METADATA_ORDER_COLUMN_NAME` quite a bit in this PR. I think we should pick a convention and stick with it.